flask_stupe/pagination.py

```diff
@@ -1,17 +1,42 @@
 import functools
 
 from flask import request
-
 from flask_stupe import pymongo
 
 __all__ = []
 
 
 if pymongo:
-    def _paginate(cursor, skip=None, limit=None, sort=None, count=True):
-        metadata = getattr(request, "metadata", None)
-        if count and isinstance(metadata, dict):
-            metadata.update(count=cursor.count())
+
+    def _get_link_header(base_url, total_count, limit, skip):
+        links = []
+        template = '<{}?limit={{limit}}&skip={{skip}}>; ' \
+                   'rel="{{rel}}"'.format(base_url)
+        prev_skip = skip - limit
+        next_skip = skip + limit
+        if (prev_skip >= 0):
+            k = "prev" if (prev_skip > limit) else "first"
+            links.append(template.format(limit=limit, skip=skip, rel=k))
+        if (next_skip < total_count):
+            k = "next" if (next_skip < (total_count - limit)) else "last"
+            links.append(template.format(limit=limit, skip=skip, rel=k))
+        return ", ".join(links)
```

Comment:
It seems that we only include `first` and `last` if we are on the edge, and thus we give inconsistent returns to the API caller. Can you return the four relatives (`first`, `prev`, `next`, `last`) in all cases?

Reply:
Good point, `first` & `last` would be nice to have in all cases indeed